### DISCUSSION TITLE
PCB Design

### DISCUSSION THREAD

**Discussion by ikonko** [acemielektron/fddEMU#9] [Ideas]
Hi,

I have finished the design of possible PCB layout for fddEMU yesterday. 
You can see the result [here](https://sindik.at/img/fddEMU_v1.png)

Alongside the design I had to create full schematic - it's [here](https://sindik.at/img/fddEMU_schematic_v1.png)

@acemielektron feel free to use it for master `README.md `update.

Kindly comment or provide improvement inputs.

Thanks.

Cheers,
Imrich

**Comment by acemielektron** [comment]
Thank you very much for your work on fddEMU, I am trying to merge atmega32u4 branch to main. I think it would be better if I use the PCB design after this merge.

**Reply by ikonko** [reply]
Pardon my ignorance but is 32U4 just a different chip on newer Arduino Nano modules or completely redesigned pinout or specs?

**Reply by acemielektron** [reply]
[Atmega32U4](https://www.microchip.com/en-us/product/ATmega32U4) is the micro-controller used in [Arduino Pro Micro](https://www.sparkfun.com/products/12640), [Arduino Leonardo](https://www.arduino.cc/en/Main/Arduino_BoardLeonardo) and [Arduino Micro](https://docs.arduino.cc/hardware/micro/) it has hardware usb. Unfortunately, unlike Atmega328 based Arduino boards, many pins of Atmega32U4 are not routed to pin headers in 32U4 based Arduino boards.

**Reply by ikonko** [reply]
Thanks for clarification.

For the purpose I'd like to use fddEMU, the upgrade to Atmega32U4 isn't the evolution I would welcome. 
My primary intention is to replace the floppy drive in multiple 8-bit computer systems including MSX platform and Sinclair plus its derivates (Didaktik, Pentagon) where DD disk support is required and existence of serial port is highly preffered because this enable direct connection from an 8-bit computer to fddEMU and easily eliminates the GUI part.

So for the time being I really plan to stay with original Arduino Nano.

**Reply by acemielektron** [reply]
fddEMU will still work on Atmega328P (Arduino Nano), in fact I merged the 32U4 branch after testing on Arduino Nano. Just that fddEMU now also work on Atmega32U4 (Arduino Pro Micro).
Arduino Nano is still the better platform for fddEMU, on Atmega32U4 [LUFA (Lightweight USB Framework for AVRs)](http://www.fourwalledcubicle.com/LUFA.php) is required for USB-Serial communication. Due to added code overhead implementing all features of fddEMU on Arduino Pro Micro is not possible.

**Comment by acemielektron** [comment]
I have finally looked at the schematic and I have two concerns:
**1)** as far as I know all odd pins should be connected to GND. In the schematic pins 29 and 31 are connected to GND, pin 33 is connected to GND through resistor R1 (value not shown), rest of the odd pins are left unconnected. In the power input side  5V input is connected to VCC through RW1 and VIN through RW2 (values not shown). 
**2)** I have previously thought about power input for fddEMU and at that time due to not having the keyed connector for floppy driver power cable, considered using the 12V input on VIN to make it foolproof. If somehow power is connected in reverse all that happens is 5V on VIN and emulator might not work, otherwise 12V on VCC kills Arduino and might even damage SD-card and screen. Of course if we have the proper keyed connector this would be a non issue. Still 12V on AMS1117 linear power regulator is close to upper limit of the regulator some diode(s) at input might help. A clearly silkscreened PCB with clear labeling both voltage and color (5V=RED, GND=BLACK, 12V=YELLOW) might also be sufficient.

**Comment by acemielektron** [comment]
I found a great article at [https://sindik.at/?p=930987](https://sindik.at/?p=930987#more-930987) documenting how the build and use fddEMU. Thanks.

**Reply by ikonko** [reply]
I wrote it with pleasure, this project deserves it 🥇

**Comment by ikonko** [comment]


**Reply by ikonko** [reply]
Hi,

thanks for you review.

I think schematic and PCB require more description about possible configurations:

- All resistors without a value represent an option, usually it's an SMD variation of a "jumper", so you can short it by wire one or use the resistor with 0 ohm value
- R**W**x are related to power - if you're unsure of voltage of you power source, you opt for RW2, if you are sure it's 5V you short RW1 - just one of two can be used
- R**O**x are related to OLED configuration. Use RO1 for 3V3 OLED display, RO2 for 5V display alternatively. Some displays have swapped `Vcc` and `GND` pins. When RO3 and RO4 are used simultaneously the standard OLED pinout is applied - GND, Vcc, SDA, SCL. Alternatively RO6 and RO5 are used for swapped Vcc and GND
- R10-16 and R110-116 (both rows has one leg connected to FDD pins **10,12,14,16**) are used for drive select configuration. Some older systems use pin 10 for **/DS0** and pin 12 for **/DS1**. In that case you short R10 and R112. Similarly you can connect **/DS1** and **/MO** signal in single drive configuration to R12 and R116 for PC-like config with crossed cable
- R**P**x resistors are 1k pullups
- R**X**x and R**T**x can be used as voltage dividers in case the signal level conversion is required. Position RX2 can be populated by Zener diode or ESD for example and RX1 can be short or small value <1k can be used for DC protection. When using MAX232 (or 3232) cable, short RX1 and RT1
- Push button **S1** is put to the side of the board and can be omitted, as not really needed by firmware
- SD module connector is split because of the stacked config of multiple interconnected PCBs
- And now a bit of elaboration of GND from host system. R1 is again a "jumper" can be used if GND is available there. PC systems might have short whole odd side of a connector, but see the schematic for external drive for system I'd like to enhance by **fddEMU** with - Didaktik Kompakt. On PC, 3 GND signals will be sufficient and there will be no showstopper to use the PCB on other systems too

[Didaktik external drive connection](https://sindik.at/img/kompakt_external_fdd.png)

**Reply by acemielektron** [reply]
Thank you, for your explanation. 
Your design allows use of either 5V or 12V input decided by the resistor (or jumper) connected. And has more configuration options. I have added it's picture and a link to this discussion thread to README.md f70aa23

**Comment by ikonko** [comment]
Finally the manufactured boards arrived and I managed to populate them and test. I definitely claim success.

The photos are in my fork :
https://github.com/ikonko/fddEMU/blob/main/images/ikonko/fddEMU_PCB1_0_top.jpg
https://github.com/ikonko/fddEMU/blob/main/images/ikonko/fddEMU_PCB1_0._back.jpg
https://github.com/ikonko/fddEMU/blob/main/images/ikonko/fddEMU_PCB1_0.side.jpg
 
The result is very compact and really sturdy

I'll be adding Gerber files soon.

**Reply by acemielektron** [reply]
> Finally the manufactured boards arrived and I managed to populate them and test. I definitely claim success.
> 
> The photos are in my fork : https://github.com/ikonko/fddEMU/blob/main/images/ikonko/fddEMU_PCB1_0_top.jpg https://github.com/ikonko/fddEMU/blob/main/images/ikonko/fddEMU_PCB1_0._back.jpg https://github.com/ikonko/fddEMU/blob/main/images/ikonko/fddEMU_PCB1_0.side.jpg
> 
> The result is very compact and really sturdy
> 
> I'll be adding Gerber files soon.

Thanks for your work on fddEMU and congratulations on your successful PCB design. Your PCB design would help people adopt fddEMU to their projects. As my perfboarded fddEMU sometimes malfunction due to bad contacts, I might build one myself.
